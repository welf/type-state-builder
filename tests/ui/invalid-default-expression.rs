use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct InvalidDefault {
    name: String,
    
    // This should be an error: invalid default expression
    #[builder(default = "this_function_does_not_exist()")]
    count: u32,
}

fn main() {
    let _instance = InvalidDefault::builder()
        .name("test".to_string())
        .build();
}