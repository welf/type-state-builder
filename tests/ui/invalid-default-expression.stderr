error[E0425]: cannot find function `this_function_does_not_exist` in this scope
 --> tests/ui/invalid-default-expression.rs:3:10
  |
3 | #[derive(TypeStateBuilder)]
  |          ^^^^^^^^^^^^^^^^ not found in this scope
  |
  = note: this error originates in the derive macro `TypeStateBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
